gslib/gcs_json_api.py

```diff
@@ -228,7 +231,15 @@ def __init__(self,
 
     self.http_base = 'https://'
     gs_json_host = config.get('Credentials', 'gs_json_host', None)
-    self.host_base = gs_json_host or 'storage.googleapis.com'
+    if context_config.get_context_config().use_client_certificate:
+      if gs_json_host:
+        raise ArgumentException(
+            '"use_client_certificate" is enabled, which sets gsutil to use the'
+            ' endpoint {}. However, a custom endpoint was also set: {}'.format(
```

Comment:
Maybe add a sentence suggesting that the user disable use_client_certificate or un-set a custom endpoint to fix the error.

Reply:
Done!